### PR TITLE
Make group_users.group_config_id NOT NULL

### PR DESCRIPTION
**Pull request by nwalters512**
Another step towards addressing #8394.

### CODE REVIEW

File: apps/prairielearn/src/migrations/20240111004522_group_users__group_config_id__not_null.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
+-- After backfilling, we still had some `group_users` rows where `group_config_id` was NULL.
+-- This is because some `groups` had a NULL `group_config_id`. We'll be fixing that separately,
+-- but we'll remove those invalid rows from `group_users` so we can add the constraint.
+DELETE FROM group_users WHERE group_config_id IS NULL;
```
Comment (nwalters512) [inline apps/prairielearn/src/migrations/20240111004522_group_users__group_config_id__not_null.sql:4]: This should be safe: I'm pretty confident that `groups` rows without a `group_config_id` are unreachable via all the joins we use to query them, and thus all `group_users` for those groups are as well.

**Comment by jonatanschroeder** [comment]
Are you planning to do this separately for `groups` and `group_users`?

**Review by jonatanschroeder — COMMENTED** [review]
Again, no change in db-types. I believe `group_config_id` is currently missing from there, so maybe add it for completeness?

File: apps/prairielearn/src/migrations/20240111005800_group_users__group_config_id__not_null.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    ALTER TABLE group_users DROP CONSTRAINT group_users_group_config_id_not_null;
+
+    RAISE EXCEPTION 'NULL row(s) exist in column, unable to add NOT NULL constraint';
+  END IF;
+END;
+$$;
+
+ALTER TABLE group_users
+ALTER COLUMN group_config_id
+SET NOT NULL;
```
Comment (nwalters512) [inline apps/prairielearn/src/migrations/20240111005800_group_users__group_config_id__not_null.sql:32]:
```suggestion
```

**Comment by nwalters512** [comment]
@jonatanschroeder `db-types` updated in 345a74c.

**Pull request merged**